spotbugs/src/main/java/edu/umd/cs/findbugs/config/CommandLine.java

```diff
@@ -342,9 +354,32 @@ private int parse(String argv[], boolean dryRun) throws IOException, HelpRequest
             }
         }
 
+        if (!dryRun && !parsingStdin && parseStdinAsOptions) {
+            String[] stdinArgv = readStdinAsArgv();
+            if (0 != stdinArgv.length) {
+                parse(stdinArgv, true, dryRun);
+            }
+        }
+
         return arg;
     }
 
+    private String[] readStdinAsArgv() throws IOException {
+        List<String> stdinArgvList = new ArrayList<String>();
+
+        BufferedReader in = UTF8.bufferedReader(System.in);
+        while (true) {
+            String s = in.readLine();
+            if (s == null) {
```

Comment:
Should this handle (ie skip) blank lines?

Reply:
Handling blank lines is probably a good idea since someone might be manually entering data into stdin or using a script.

Your comment about using Appendable and getAppendableInputString is already marked resolved so I can't answer there, but I think you're over-designing it.   This is the API for setting up the ant task.  It's rarely going to be used, and what's already there works well enough.